lib/access_token_generator.rb

```diff
@@ -0,0 +1,21 @@
+module AccessTokenGenerator
+  # generate an access token from a service account JSON keyfile
+  #
+  # * *params*
+  #   - +service_account+ (Pathname) => path to service account JSON keyfile
+  #
+  # * *returns*
+  #   - (Hash) => OAuth2 access token hash, with the following attributes
+  #     - +access_token+ (String) => OAuth2 access token
+  #     - +expires_in+ (Integer) => duration of token, in seconds
+  #     - +token_type+ (String) => type of access token (e.g. 'Bearer')
+  def generate_access_token(service_account)
```

Comment:
I'd make this a class method `self.generate_access_token`, and have it take the scopes as an argument.  that makes it clearer what this method depends on

Reply:
5eb5494 - this is renamed to `ServiceAccountManager`, and since it is only ever used via `extend` in other classes, this means that it is in fact a class method.  

This is the only way to get this to work as using `include` provides access as instance (not class) methods, and you can't mix the two without lots of chicanery that breaks other things.

Reply:
I think I'm fuzzy on the difference between this and the google_service_client.rb. Should this be moved into that file so we centralize google token creation/refresh?